src/hdmf/data_utils.py

```diff
@@ -2,9 +2,12 @@
 from abc import ABCMeta, abstractmethod
 from collections.abc import Iterable
 from warnings import warn
+from typing import Tuple
+from itertools import product, chain
 
 import h5py
 import numpy as np
+import psutil
```

Comment:
if `psutil` is used only in `GenericDataChunkIterator`, can we make it an optional import? Since HDMF is a library, it would be better to keep it relatively lightweight in terms of dependencies. 

(To make it an optional import, import it when first used and raise an informative error if the package is not found. Also describe the optional import in the installation requirements and readme.)